[cinatra, tanakh-cmdline] Add new ports

EmployeeNo427 opened the pull request:
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html).
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

EmployeeNo427 committed:
[cinatra, cmdline] Add new ports

EmployeeNo427 committed:
[cinatra, cmdline] add version

EmployeeNo427 committed:
Update ci.baseline.txt to mark cinatra as unsupported on android and uwp

Cheney-W commented:
This port has the following issues:
1. Its dependencies are not being used at all, as there is no related code in the source's CMakeLists.txt.
2. The headers provided by the source contain many calls to headers in the same or lower-level directories. If these calls are to use files from those dependencies, the current file structure does not match.
3. You have used some deprecated functions, and vcpkg's policy is not to build examples.

EmployeeNo427 committed:
Update cinatra/portfile.cmake and cinatra/vcpkg.json to include dependencies as part of the cinatra installation

EmployeeNo427 committed:
Update version of ports/cinatra/vcpkg.json to reflect changes

EmployeeNo427 committed:
Update with vcpkg x-add-version cinatra

Cheney-W commented on ports/cinatra/portfile.cmake:
```diff
@@ -0,0 +1,43 @@
+    HEAD_REF master
+)
+
+vcpkg_configure_cmake(
+    SOURCE_PATH ${SOURCE_PATH}
+    PREFER_NINJA
+    OPTIONS
+        -DBUILD_UNIT_TESTS=OFF
+        -DBUILD_EXAMPLES=ON
+        -DBUILD_PRESS_TOOL=ON
+)
+
+vcpkg_install_cmake()
```
```suggestion
vcpkg_cmake_configure(
    SOURCE_PATH "${SOURCE_PATH}"
    OPTIONS
        -DBUILD_UNIT_TESTS=OFF
        -DBUILD_EXAMPLES=OFF
        -DBUILD_PRESS_TOOL=ON
)

vcpkg_cmake_install()

```

Cheney-W commented on ports/cinatra/portfile.cmake:
```diff
@@ -0,0 +1,43 @@
+    SHA512 55bba51bb6190b76bc6415d3c36e82daeb5f174f1cb10490951cf05863f55653a6d68ffd3b66c5b3b2ebb6e68a7a84c090e973a6718f3f2a5849b04330e4b180
+    HEAD_REF master
+)
+
```
```suggestion

set(VCPKG_BUILD_TYPE "release") # header-only port

```

Cheney-W commented on ports/cinatra/portfile.cmake:
```diff
@@ -0,0 +1,43 @@
+# Copy the entire include directory
+file(COPY ${SOURCE_PATH}/include DESTINATION ${CURRENT_PACKAGES_DIR})
+
+# Copy example web content
+file(COPY ${SOURCE_PATH}/example/www DESTINATION ${CURRENT_PACKAGES_DIR}/tools/cinatra/examples)
+
```
```suggestion
```

Cheney-W commented on ports/cinatra/vcpkg.json:
```diff
@@ -0,0 +1,7 @@
+  "version-date": "2024-05-15",
+  "description": "Cinatra is a high performance HTTP framework with modern C++ features.",
+  "homepage": "https://github.com/qicosmos/cinatra",
+  "license": "MIT"
```
```suggestion
  "license": "MIT",
  "dependencies": [
    {
      "name": "vcpkg-cmake",
      "host": true
    }
  ]
```

Cheney-W commented on ports/cinatra/vcpkg.json:
```diff
@@ -0,0 +1,7 @@
+{
+  "name": "cinatra",
+  "version-date": "2024-05-15",
```
```suggestion
  "version": "0.9.1",
```

Cheney-W commented on scripts/ci.baseline.txt:
```diff
@@ -190,6 +190,12 @@ chromium-base:x64-linux=skip
 #  because the first attempt to build it fails, but subsequent attempts succeed
 chromium-base:x64-osx=skip
 
+cinatra:x64-android=fail
+cinatra:arm-neon-android=fail
+cinatra:arm64-android=fail
+cinatra:x64-uwp=fail
+cinatra:arm64-uwp=fail
+
```
```suggestion
```
If this port does not support Android and UWP, you can add `"supports": "!uwp & !android"` on the line following the `license` statement in `vcpkg.json`.

Cheney-W commented on ports/cinatra/portfile.cmake:
```diff
@@ -0,0 +1,43 @@
+endif()
+
+vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE")
+vcpkg_copy_pdbs()
+
+# Cleanup unwanted directories
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/bin")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/bin")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug")
+file(REMOVE_RECURSE "${SOURCE_PATH}/.git")
```
```suggestion
```

Cheney-W commented on ports/cinatra/portfile.cmake:
```diff
@@ -0,0 +1,43 @@
+file(COPY ${SOURCE_PATH}/example/www DESTINATION ${CURRENT_PACKAGES_DIR}/tools/cinatra/examples)
+
+# Copy executables to the tools directory
+if(EXISTS "${CURRENT_PACKAGES_DIR}/bin/cinatra_example.exe")
+    file(COPY "${CURRENT_PACKAGES_DIR}/bin/cinatra_example.exe"
+         DESTINATION "${CURRENT_PACKAGES_DIR}/tools/cinatra")
+    file(COPY "${CURRENT_PACKAGES_DIR}/bin/cinatra_press_tool.exe"
+         DESTINATION "${CURRENT_PACKAGES_DIR}/tools/cinatra")
+endif()
```
Only the copy of `cinatra_press_tool.exe` needs to be retained here.

EmployeeNo427 committed:
Update ports/cinatra/portfile.cmake

Co-authored-by: Cheney Wang <38240633+Cheney-W@users.noreply.github.com>

EmployeeNo427 committed:
Update ports/cinatra/portfile.cmake

Co-authored-by: Cheney Wang <38240633+Cheney-W@users.noreply.github.com>

EmployeeNo427 committed:
Update ports/cinatra/portfile.cmake

Co-authored-by: Cheney Wang <38240633+Cheney-W@users.noreply.github.com>

EmployeeNo427 committed:
Update ports/cinatra/vcpkg.json

Co-authored-by: Cheney Wang <38240633+Cheney-W@users.noreply.github.com>

EmployeeNo427 committed:
Update ports/cinatra/vcpkg.json

Co-authored-by: Cheney Wang <38240633+Cheney-W@users.noreply.github.com>

EmployeeNo427 committed:
Update ports/cinatra and scripts/ci.baseline.txt

EmployeeNo427 commented:
@Cheney-W Thank you very much for the detailed feedback! I have applied all the suggestions.

Cheney-W commented:
0.9.1 is declared with ce7e7fba41dc0718831bc3ce20644051f46a6878, but the local port has a different SHA 91af211a6a62396620de86d320f075490e0ea302.
Please update the port's version fields and then run:
vcpkg x-add-version cinatra
git add versions
git commit -m "Update version database"
to add the new version.
To attempt to resolve all errors at once, run:
vcpkg x-add-version --all

Cheney-W committed:
Update cinatra.json

EmployeeNo427 commented:
> 0.9.1 is declared with ce7e7fba41dc0718831bc3ce20644051f46a6878, but the local port has a different SHA 91af211a6a62396620de86d320f075490e0ea302. Please update the port's version fields and then run: vcpkg x-add-version cinatra git add versions git commit -m "Update version database" to add the new version. To attempt to resolve all errors at once, run: vcpkg x-add-version --all

Thank you for the feedback. I tried to do something like that, but I encountered an error:
```
PS D:\opensourcework\vcpkgstuff\vcpkg> git add ports/cinatra
PS D:\opensourcework\vcpkgstuff\vcpkg> git commit -m "Update cinatra/portfile.cmake and cinatra/vcpkg.json to include dependencies as part of the cinatra installation"
[add-cinatra-port fca3554da] Update cinatra/portfile.cmake and cinatra/vcpkg.json to include dependencies as part of the cinatra installation
 2 files changed, 5 insertions(+), 25 deletions(-)
PS D:\opensourcework\vcpkgstuff\vcpkg> git push -u origin add-cinatra-port
Enumerating objects: 11, done.
Counting objects: 100% (11/11), done.
Delta compression using up to 8 threads
Compressing objects: 100% (6/6), done.
Writing objects: 100% (6/6), 618 bytes | 618.00 KiB/s, done.
Total 6 (delta 4), reused 0 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (4/4), completed with 4 local objects.
To https://github.com/EmployeeNo427/vcpkg.git
   2fdf2e5ed..fca3554da  add-cinatra-port -> add-cinatra-port
branch 'add-cinatra-port' set up to track 'origin/add-cinatra-port'.
PS D:\opensourcework\vcpkgstuff\vcpkg> vcpkg x-add-version cinatra
error: cinatra is not properly formatted
Run `vcpkg format-manifest "D:\opensourcework\vcpkgstuff\vcpkg\ports\cinatra\vcpkg.json"` to format the file
Don't forget to commit the result!

PS D:\opensourcework\vcpkgstuff\vcpkg> vcpkg format-manifest "D:\opensourcework\vcpkgstuff\vcpkg\ports\cinatra\vcpkg.json
>> ^C
PS D:\opensourcework\vcpkgstuff\vcpkg> vcpkg format-manifest "D:\opensourcework\vcpkgstuff\vcpkg\ports\cinatra\vcpkg.json"
Succeeded in formatting the manifest files.
PS D:\opensourcework\vcpkgstuff\vcpkg> vcpkg x-add-version cinatra
warning: there are uncommitted changes for cinatra
error: checked-in files for cinatra have changed but the version was not updated
version: 0.9.1
old SHA: ce7e7fba41dc0718831bc3ce20644051f46a6878
new SHA: b6989160d51c1b9ce67e472c1b2de2ae15d5f9c9
Did you remember to update the version or port version?
Use --overwrite-version to bypass this check
***No files were updated***
PS D:\opensourcework\vcpkgstuff\vcpkg> vcpkg format-manifest "D:\opensourcework\vcpkgstuff\vcpkg\ports\cinatra\vcpkg.json"
Succeeded in formatting the manifest files.
PS D:\opensourcework\vcpkgstuff\vcpkg> vcpkg x-add-version cinatra
warning: there are uncommitted changes for cinatra
error: checked-in files for cinatra have changed but the version was not updated
version: 0.9.1
old SHA: ce7e7fba41dc0718831bc3ce20644051f46a6878
new SHA: b6989160d51c1b9ce67e472c1b2de2ae15d5f9c9
Did you remember to update the version or port version?
Use --overwrite-version to bypass this check
***No files were updated***
PS D:\opensourcework\vcpkgstuff\vcpkg>
```

EmployeeNo427 commented:
> 0.9.1 is declared with ce7e7fba41dc0718831bc3ce20644051f46a6878, but the local port has a different SHA 91af211a6a62396620de86d320f075490e0ea302. Please update the port's version fields and then run: vcpkg x-add-version cinatra git add versions git commit -m "Update version database" to add the new version. To attempt to resolve all errors at once, run: vcpkg x-add-version --all

@Cheney-W Thanks! For clarification, I need to **manually** change "\vcpkg\versions\c-\cinatra.json" from:
```
{
  "versions": [
    {
      "git-tree": "f14ed90b0ac73f6a970d7688329dfbabe7b08a58",
      "version-date": "2024-05-15",
      "port-version": 0
    },
    {
      "git-tree": "ce7e7fba41dc0718831bc3ce20644051f46a6878",
      "version": "0.9.1",
      "port-version": 0
    }
  ]
}
```
to:
```
{
  "versions": [
    {
      "git-tree": "f14ed90b0ac73f6a970d7688329dfbabe7b08a58",
      "version-date": "2024-05-15",
      "port-version": 0
    },
    {
      "git-tree": "91af211a6a62396620de86d320f075490e0ea302",
      "version": "0.9.1",
      "port-version": 0
    }
  ]
}
```
And then run the commands?

Cheney-W commented:
There should be only one record in the `\vcpkg\versions\c-\cinatra.json` file.
When you do any change in the cinatra port and submit a commit, you should run below commands:
```
.\vcpkg x-add-version cinatra --overwrite-version
git add .
git commit --amend --no-edit
git push
```
Also, I noticed that `cinatra_press_tool.exe` is stored in `${CURRENT_PACKAGES_DIR}/include`, so you won't find it under `${CURRENT_PACKAGES_DIR}/bin`.

EmployeeNo427 committed:
Update version database

EmployeeNo427 committed:
Update version database

EmployeeNo427 committed:
Resolve merge conflict in versions/c-/cinatra.json

Cheney-W commented on ports/cinatra/portfile.cmake:
```diff
@@ -0,0 +1,32 @@
+
+# Copy executables to the tools directory
+if(EXISTS "${CURRENT_PACKAGES_DIR}/include/cinatra_press_tool.exe")
+    file(COPY "${CURRENT_PACKAGES_DIR}/include/cinatra_press_tool.exe"
+         DESTINATION "${CURRENT_PACKAGES_DIR}/tools/cinatra")
```
Please remove the `include/cinatra_press_tool.exe` when it is copied to `tools/cinatra`.

EmployeeNo427 replied:
Thanks!

EmployeeNo427 committed:
Update ports/cinatra/portfile.cmake

Co-authored-by: Cheney Wang <38240633+Cheney-W@users.noreply.github.com>

EmployeeNo427 committed:
Remove include/cinatra_press_tool.exe after copying to tools/cinatra

Cheney-W commented on ports/cinatra/portfile.cmake:
```diff
@@ -0,0 +1,32 @@
+vcpkg_cmake_install()
+
+# Copy the entire include directory
+file(COPY ${SOURCE_PATH}/include DESTINATION ${CURRENT_PACKAGES_DIR})
```
The current header file storage structure may cause conflicts with header files from other ports. Therefore, I suggest moving all the contents under the include directory to `${CURRENT_PACKAGES_DIR}/include/cinatra` instead of `${CURRENT_PACKAGES_DIR}/include/`.

EmployeeNo427 replied:
Thank you very much!

EmployeeNo427 committed:
Move include contents to /include/cinatra to avoid conflicts

dg0yt reviewed (commented):
vcpkg has asio, boost-asio, and async-simple. Shouldn't this port use the corresponding vcpkg ports instead of vendored copies?

EmployeeNo427 committed:
Use vcpkg_copy_tools to handle cinatra_press_tool

EmployeeNo427 commented:
> vcpkg has asio, boost-asio, and async-simple. Shouldn't this port use the corresponding vcpkg ports instead of vendored copies?

Thank you for the suggestion! I initially tried to incorporate the vcpkg ports for dependencies, but it seems to require reconfiguring the library.

https://github.com/microsoft/vcpkg/pull/38712#issuecomment-2112180645

EmployeeNo427 committed:
Use vcpkg_copy_tools to handle cinatra_press_tool

EmployeeNo427 committed:
Use vcpkg_copy_tools to handle cinatra_press_tool

dg0yt commented:
> I initially tried to incorporate the vcpkg ports for dependencies, but it seems to require reconfiguring the library.

This might be a necessary task for integrating it into vcpkg, so that downstream apps don't end up with different definitions of the same functions.

dg0yt commented on ports/cinatra/portfile.cmake:
```diff
@@ -0,0 +1,36 @@
+vcpkg_copy_tools(
+    TOOL_NAMES cinatra_press_tool
+    SEARCH_DIRS "${CURRENT_PACKAGES_DIR}/include/cinatra"
+    DESTINATION "${CURRENT_PACKAGES_DIR}/tools/cinatra"
```
```suggestion
```
This is the default.

Cheney-W replied:
The `AUTO_CLEAN` of `vcpkg_copy_tools` can only delete the tool under `${CURRENT_PACKAGES_DIR}/bin`, so `${CURRENT_PACKAGES_DIR}/include/cinatra_press_tool.exe` still needs to be deleted manually here.

dg0yt commented on ports/cinatra/portfile.cmake:
```diff
@@ -0,0 +1,36 @@
+# Copy the entire include directory to ${CURRENT_PACKAGES_DIR}/include/cinatra
+file(COPY ${SOURCE_PATH}/include DESTINATION ${CURRENT_PACKAGES_DIR}/include/cinatra)
+
+# Copy executables to the tools directory using vcpkg_copy_tools
```
```suggestion
```
`vcpkg_copy_tools` is expressive enough.

dg0yt commented:
What does  `cinatra_press_tool` do? 
It is unexpected for a "header-only" port to install tools, and it is unexpected that it installs to `include`.

EmployeeNo427 committed:
Revert to the crude method to handle cinatra_press_tool for now

EmployeeNo427 committed:
Use vcpkg_copy_tools and manually remove cinatra_press_tool from include

EmployeeNo427 commented:
There are some build errors I encountered with vcpkg_copy_tools; it is referring to the bin folder.

EmployeeNo427 commented:
@Cheney-W and @dg0yt

Thank you for your feedback and suggestions! I have reverted to using a combination of `vcpkg_copy_tools` and manually removing `cinatra_press_tool.exe` from the `include` directory to handle the tool. However, from looking at the GitHub checks, it is still referring to the `bin` folder for some reason and encountering build errors.

Additionally, I noticed there are ongoing discussions about the proper handling and integration of dependencies. Could you please provide guidance on the following points?

1. Is the current approach of using `vcpkg_copy_tools` combined with the manual removal of `cinatra_press_tool.exe` acceptable, or is there a better method you would recommend?
2. What is the best way to handle the integration of dependencies to ensure compatibility and avoid issues with downstream applications? Previously, I updated `cinatra/portfile.cmake` and `cinatra/vcpkg.json` to include dependencies as part of the cinatra installation instead of through vcpkg, due to how cinatra itself is structured, as @Cheney-W mentioned:
    ```
    Its dependencies are not being used at all, as there is no related code in the source's CMakeLists.txt.
    The headers provided by the source contain many calls to headers in the same or lower-level directories. If these calls are to use files from those dependencies, the current file structure does not match.
    ```

Again, thank you very much for your assistance on these matters!

EmployeeNo427 committed:
Update cinatra portfile to handle header files and tools

EmployeeNo427 committed:
Delete vcpkg_cmake_install()

EmployeeNo427 committed:
Delete vcpkg_copy_tools()

EmployeeNo427 committed:
Updated portfile.cmake to copy header files to the include directory and removed unnecessary handling of cinatra_press_tool

Cheney-W commented on ports/cinatra/vcpkg.json:
```diff
@@ -0,0 +1,14 @@
+  "description": "Cinatra is a high performance HTTP framework with modern C++ features.",
+  "homepage": "https://github.com/qicosmos/cinatra",
+  "license": "MIT",
+  "supports": "!uwp & !android",
+  "dependencies": [
+    {
+      "name": "vcpkg-cmake",
+      "host": true
+    }
+  ]
```
```suggestion
  "supports": "!uwp & !android"
```

Cheney-W commented on ports/cinatra/portfile.cmake:
```diff
@@ -0,0 +1,25 @@
+vcpkg_install_copyright(
+    FILE_LIST "${SOURCE_PATH}/LICENSE"
+)
+
+
+
+
+
+
+
+
```
```suggestion
```

Cheney-W commented on ports/cinatra/portfile.cmake:
```diff
@@ -0,0 +1,25 @@
+set(VCPKG_BUILD_TYPE "release") # Header-only library
+
+# Copy all header files directly into the include directory of the package
+file(COPY ${SOURCE_PATH}/include DESTINATION ${CURRENT_PACKAGES_DIR}/include)
```
```suggestion
file(COPY "${SOURCE_PATH}/include" DESTINATION "${CURRENT_PACKAGES_DIR}/include")
file(RENAME "${CURRENT_PACKAGES_DIR}/include/include" "${CURRENT_PACKAGES_DIR}/include/cinatra")
```

Cheney-W commented on ports/cinatra/portfile.cmake:
```diff
@@ -0,0 +1,25 @@
+    HEAD_REF master
+)
+
+set(VCPKG_BUILD_TYPE "release") # Header-only library
+
```
```suggestion
```

Cheney-W commented on ports/tanakh-cmdline/portfile.cmake:
```diff
@@ -0,0 +1,12 @@
+
+vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE")
+
+file(INSTALL ${SOURCE_PATH}/cmdline.h
+     DESTINATION ${CURRENT_PACKAGES_DIR}/include)
```
```suggestion
file(INSTALL "${SOURCE_PATH}/cmdline.h"
     DESTINATION "${CURRENT_PACKAGES_DIR}/include/tanakh-cmdline"
```

EmployeeNo427 replied:
Thank you for taking the time to review! Some of these changes may have caused the CI build to fail, but I am not sure why. Could you help me pinpoint the issues and advise me on how to proceed?

EmployeeNo427 committed:
Update ports/cinatra/vcpkg.json

Co-authored-by: Cheney Wang <38240633+Cheney-W@users.noreply.github.com>

EmployeeNo427 committed:
Update ports/cinatra/portfile.cmake

Co-authored-by: Cheney Wang <38240633+Cheney-W@users.noreply.github.com>

EmployeeNo427 committed:
Update ports/cinatra/portfile.cmake

Co-authored-by: Cheney Wang <38240633+Cheney-W@users.noreply.github.com>

EmployeeNo427 committed:
Update ports/cinatra/portfile.cmake

Co-authored-by: Cheney Wang <38240633+Cheney-W@users.noreply.github.com>

EmployeeNo427 committed:
Update ports/tanakh-cmdline/portfile.cmake

Co-authored-by: Cheney Wang <38240633+Cheney-W@users.noreply.github.com>

EmployeeNo427 committed:
Fix missing ')' in tanakh-cmdline portfile.cmake

EmployeeNo427 commented:
Hi @Cheney-W, thank you for your previous review!
Are there any remaining changes needed? I noticed there's one requested change, but I couldn't find any specific details about it.

Cheney-W committed:
Update cinatra.json

JavierMatosD commented on ports/cinatra/portfile.cmake:
```diff
@@ -0,0 +1,16 @@
+vcpkg_from_github(
```
```suggestion
set(VCPKG_BUILD_TYPE "release") # header-only port
vcpkg_from_github(
```

JavierMatosD commented on ports/tanakh-cmdline/portfile.cmake:
```diff
@@ -0,0 +1,14 @@
+vcpkg_from_github(
```
```suggestion
set(VCPKG_BUILD_TYPE "release") # header-only port
vcpkg_from_github(
```

EmployeeNo427 committed:
Update ports/cinatra/portfile.cmake

Co-authored-by: Javier Matos Denizac <javier.matosd@gmail.com>

EmployeeNo427 committed:
Update ports/tanakh-cmdline/portfile.cmake

Co-authored-by: Javier Matos Denizac <javier.matosd@gmail.com>

EmployeeNo427 committed:
Refresh version database for cinatra & tanakh-cmdline after header-only tweak

dg0yt commented on ports/cinatra/portfile.cmake:
```diff
@@ -0,0 +1,17 @@
+    HEAD_REF master
+)
+
+# Copy all header files directly into the include directory of the package
+file(COPY "${SOURCE_PATH}/include" DESTINATION "${CURRENT_PACKAGES_DIR}/include")
+file(RENAME "${CURRENT_PACKAGES_DIR}/include/include" "${CURRENT_PACKAGES_DIR}/include/cinatra")
```
```suggestion
file(COPY "${SOURCE_PATH}/include/" DESTINATION "${CURRENT_PACKAGES_DIR}/include/cinatra")
```

dg0yt commented on ports/cinatra/portfile.cmake:
```diff
@@ -0,0 +1,17 @@
+file(COPY "${SOURCE_PATH}/include" DESTINATION "${CURRENT_PACKAGES_DIR}/include")
+file(RENAME "${CURRENT_PACKAGES_DIR}/include/include" "${CURRENT_PACKAGES_DIR}/include/cinatra")
+
+# Install the license file(s)
+vcpkg_install_copyright(
+    FILE_LIST "${SOURCE_PATH}/LICENSE"
+)
```
```suggestion
# Install the license file(s)
vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE")
```

dg0yt reviewed (commented):
The include dir still needs devendoring:
IIUC there are copies of tanakh-cmdline and of boost-asio.

EmployeeNo427 committed:
Update ports/cinatra/portfile.cmake

Co-authored-by: Kai Pastor <dg0yt@darc.de>

EmployeeNo427 committed:
Update ports/cinatra/portfile.cmake

Co-authored-by: Kai Pastor <dg0yt@darc.de>

EmployeeNo427 committed:
Remove vendored deps; install only Cinatra headers

EmployeeNo427 committed:
Format cinatra manifest

EmployeeNo427 committed:
Refresh version database for cinatra

Cheney-W commented on ports/cinatra/portfile.cmake:
```diff
@@ -0,0 +1,20 @@
+)
+
+# Install only Cinatra’s own headers—not vendored dependencies
+file(INSTALL
+    "${SOURCE_PATH}/include/cinatra"
+    DESTINATION "${CURRENT_PACKAGES_DIR}/include/cinatra"
+)
+file(INSTALL
+    "${SOURCE_PATH}/include/cinatra.hpp"
+    DESTINATION "${CURRENT_PACKAGES_DIR}/include"
+)
```
```suggestion
file(INSTALL
    "${SOURCE_PATH}/include/cinatra"
    "${SOURCE_PATH}/include/cinatra.hpp"
    DESTINATION "${CURRENT_PACKAGES_DIR}/include"
)
```
The original command generates the following structure:
```
include/cinatra.hpp
include/cinatra/cinatra/xxx
```
This causes an error in `cinatra.hpp` for the line `#include "cinatra/coro_http_client.hpp"`.

EmployeeNo427 committed:
Update ports/cinatra/portfile.cmake

Co-authored-by: Cheney Wang <38240633+Cheney-W@users.noreply.github.com>

EmployeeNo427 committed:
Refresh version database for cinatra
